ironfish/src/rpc/adapters/tlsAdapter.ts

```diff
@@ -37,6 +42,15 @@ export class RpcTlsAdapter extends RpcSocketAdapter {
   protected async getTlsOptions(): Promise<tls.TlsOptions> {
     const nodeKeyExists = await this.fileSystem.exists(this.nodeKeyPath)
     const nodeCertExists = await this.fileSystem.exists(this.nodeCertPath)
+    const rpcAuthToken = this.node.internal.get('rpcAuthToken')
+
+    if (!rpcAuthToken || rpcAuthToken === '') {
+      this.logger.debug(
+        `Missing RPC Auth token in internal.json config. Automatically generating auth token.`,
+      )
+      this.node.internal.set('rpcAuthToken', uuid())
+      await this.node.internal.save()
+    }
```

Comment:
If we add another adapter that uses the auth, and don't mount the TlsAdapter, then authentication will break because this is the only place that generates a token. We should do it in a more general place like the RpcServer because that's the only place that consumes the token.

Reply:
If we add another adapter, it should write the token generator code if they use auth token.

Reply:
I'm unsure of what you mean, do you mean you'll copy paste this logic into each adapter that needs to ensure that it has a token?